src/qibolab/result.py

```diff
@@ -114,20 +114,15 @@ class StateResults:
     Associated with AcquisitionType.DISCRIMINATION and AveragingMode.SINGLESHOT
     """
 
-    def __init__(self, states: np.ndarray = np.array([]), nshots=None):
-        self.states: Optional[npt.NDArray[np.uint32]] = (
-            states.reshape(states.shape[0] // nshots, nshots) if nshots else states
-        )
-
-    @property
-    def states(self):
-        return self._states
+    states: npt.NDArray[np.uint32]
+    nshots: Optional[int] = None
 
-    @states.setter
-    def states(self, values):
-        if not np.all((values >= 0) & (values <= 1)):
+    def __post_init__(self, states: np.ndarray = np.array([]), nshots=None):
+        if not np.all((self.states >= 0) & (self.states <= 1)):
             raise ValueError("Probability wrong")
-        self._states = values
+        if nshots is not None:
+            # TODO: Ask what is this reshape?
```

Comment:
I think it is to better deal with sweepers, so that it automatically splits the states from different sequences. However now that I think about it, this will probably change things also in qibocal and elsewhere @Jacfomg